### PR TITLE
Prettier

### PR DESCRIPTION
**Pull request by vlad-zhukov**
Removed standard in favour of prettier.

Supersedes #263. Unlike that PR I've decided to drop standard completely because it doesn't make any noticeable difference. Also I decided to not setup on commit hook as it's never worked good for me, if someone finds it useful send another PR.

@andywer @zcei @dmitmel @jvanbruegge

### CODE REVIEW

**Comment by andywer** [comment]
Prettier is a good idea!

Just two questions that we should discuss together (didn't we have an ongoing discussion somewhere already?):
1. Wouldn't it be best to still stick to a widespread code style, prettier or not?
2. I think dropping eslint is a mistake, since prettier is just for code style, while eslint's capabilities go beyond that (and we already use them)? We need it to detect unused variables, broken references and similar.

**Comment by zcei** [comment]
+1 for not adding a pre-commit hook. That only made me use `--no-verify` more often than not :D 

I also agree with Andy in that we still should have an ESLint setup for more code semantics. Could either stick with `standard` or try to align Prettier with ESLint by incorporating an ESLint setup, maybe with `standard` preset to keep changes to a minimum.

**Comment by jvanbruegge** [comment]
Why no pre-commit hook? Prettier runs fast and this keeps the code formatted. I understand that with linters as you do WIP commits, but prettier is not really an issue.

**Comment by vlad-zhukov** [comment]
Right, nice that you've reminded me of `eslint`, I forgot about that one 😄. Can I set it up in the next PR?

@andywer 
1. I've configured prettier to be as close to standard as possible for now (but changes are inevitable). The great thing about the prettier is that it enforces the way how to read the code, but not how write it. With it I can write code the hell I want (messy whitespace, long ass lines and so on) to a point when I can't even read it, then simply run `yarn format`, and my code is fine again.

**Comment by dmitmel** [comment]
@jvanbruegge @zcei What about running `prettier` on CI?

**Comment by andywer** [comment]
Yeah, might make most sense to do it in this PR. Could do it in a follow-up PR, but then we should also create an issue for that, so we definitely don't forget about it...

File: package.json [hunk 1]
```diff
@@ -1,14 +1,21 @@
 {
   "private": true,
   "scripts": {
-    "test": "lerna run test"
+    "test": "yarn prettier --debug-check && lerna run test",
+    "format": "yarn prettier --write",
+    "prettier": "prettier \"**/*.{js,ts,tsx}\" --loglevel warn"
```
[the remark this answers is not in the document]
Comment (vlad-zhukov) [reply package.json]: There is a reason why I wrap globs with double quotes in my projects. If I recall it correctly single quotes don't work in certain shell(s) under Windows. I am not able to test it anytime soon, so I will make the requested change now and revert it back if there will be a problem.

File: package.json [hunk 2]
```diff
@@ -1,14 +1,24 @@
 {
   "private": true,
   "scripts": {
-    "test": "lerna run test"
+    "test": "yarn prettier --list-different && yarn lint && lerna run test",
+    "lint": "eslint \"**/*.js\"",
+    "format": "yarn prettier --write",
+    "prettier": "prettier \"**/*.{js,ts,tsx}\" --loglevel warn"
```
Comment (vlad-zhukov) [inline package.json]: Double quotes are required on Windows. With single quotes both prettier and eslint couldn't fine any files.

**Comment by vlad-zhukov** [comment]
Added `eslint`! 🎉

**Pull request merged**